packages/xrpl/test/integration/transactions/checkCash.test.ts

```diff
@@ -0,0 +1,78 @@
+import { assert } from 'chai'
+import { CheckCreate, CheckCash } from 'xrpl-local'
+
+import serverUrl from '../serverUrl'
+import {
+  setupClient,
+  teardownClient,
+  type XrplIntegrationTestContext,
+} from '../setup'
+import { generateFundedWallet, testTransaction } from '../utils'
+
+// how long before each test case times out
+const TIMEOUT = 20000
+
+describe('CheckCash', () => {
+  let testContext: XrplIntegrationTestContext
+
+  beforeEach(async () => {
+    testContext = await setupClient(serverUrl)
+  })
+  afterEach(async () => teardownClient(testContext))
+
+  it(
+    'base',
+    async () => {
+      const wallet2 = await generateFundedWallet(testContext.client)
+      const amount = '500'
+
+      const setupTx: CheckCreate = {
+        TransactionType: 'CheckCreate',
+        Account: testContext.wallet.classicAddress,
+        Destination: wallet2.classicAddress,
+        SendMax: amount,
+      }
+
+      await testTransaction(testContext.client, setupTx, testContext.wallet)
+
+      // get check ID
+      const response1 = await testContext.client.request({
+        command: 'account_objects',
+        account: testContext.wallet.classicAddress,
+        type: 'check',
+      })
+      assert.lengthOf(
+        response1.result.account_objects,
+        1,
+        'Should be exactly one check on the ledger',
+      )
+      const checkId = response1.result.account_objects[0].index
+
+      // actual test - cash the check
+      const tx: CheckCash = {
+        TransactionType: 'CheckCash',
+        Account: wallet2.classicAddress,
+        CheckID: checkId,
+        Amount: amount,
+      }
+
+      await testTransaction(testContext.client, tx, wallet2, {
+        count: 5,
```

Comment:
Might be worth a comment explaining why so many retries are necessary

Reply:
Done

Reply:
Is this still necessary now that tests are not being run concurrently?